### PR TITLE
Change OpenTelemetry timer unit to seconds [PLEN-44]

### PR DESCRIPTION
**Pull request by nicu-da**
This also impacts the buckets used for the timer histogram.

### CODE REVIEW

**Comment by lucb-da** [comment]
I expect the test values for latency in [AkkaHttpMetricsSpec](https://github.com/digital-asset/daml/blob/main/observability/akka-http-metrics/src/test/suite/scala/com/daml/metrics/akkahttp/AkkaHttpMetricsSpec.scala#L633) will have to be updated (and the type switched to `double`).

**Comment by nicu-da** [comment]
> I expect the test values for latency in [AkkaHttpMetricsSpec](https://github.com/digital-asset/daml/blob/main/observability/akka-http-metrics/src/test/suite/scala/com/daml/metrics/akkahttp/AkkaHttpMetricsSpec.scala?rgh-link-date=2022-11-16T11%3A00%3A31Z#L633) will have to be updated (and the type switched to `double`).

Actually no, as the in memory impl is actually based on millis not seconds.

File: ledger/metrics/src/main/scala/com/daml/metrics/OpenTelemetryMeterOwner.scala [hunk 1]
```diff
@@ -47,23 +48,35 @@ case class OpenTelemetryMeterOwner(enabled: Boolean, reporter: Option[MetricsRep
...
       )
+  }
+
+  private def histogramSelectorEndingWith(endingWith: String) = {
+    InstrumentSelector
+      .builder()
+      .setType(InstrumentType.HISTOGRAM)
+      .setName((t: String) => t.endsWith(endingWith))
+      .build()
+  }
```
Comment (miklos-da) [inline ledger/metrics/src/main/scala/com/daml/metrics/OpenTelemetryMeterOwner.scala:69]: Nit: This and the below are single-statement functions so you could remove the surrounding curly braces:
```suggestion
  private def histogramSelectorEndingWith(endingWith: String) =
    InstrumentSelector
      .builder()
      .setType(InstrumentType.HISTOGRAM)
      .setName((t: String) => t.endsWith(endingWith))
      .build()
```

File: ledger/metrics/src/main/scala/com/daml/metrics/api/opentelemetry/OpenTelemetryFactory.scala [hunk 2]
```diff
@@ -170,6 +179,17 @@ case class OpenTelemetryTimer(name: String, histogram: LongHistogram, timerConte
       context: MetricsContext
   ): Unit = update(duration.toNanos, TimeUnit.NANOSECONDS)
 }
+object OpenTelemetryTimer {
+
+  private[opentelemetry] val TimerUnit: String = "seconds"
+  val TimerUnitAndSuffix: MetricName = MetricName("duration", TimerUnit)
+
+  private val NanoSecondsInASecond = 1_000_000_000
```
Comment (miklos-da) [inline ledger/metrics/src/main/scala/com/daml/metrics/api/opentelemetry/OpenTelemetryFactory.scala:187]: Nit: For consistency:
```suggestion
  private val NanosecondsInASecond = 1_000_000_000
```

File: ledger/metrics/src/main/scala/com/daml/metrics/api/opentelemetry/OpenTelemetryFactory.scala [hunk 3]
```diff
@@ -170,6 +179,17 @@ case class OpenTelemetryTimer(name: String, histogram: LongHistogram, timerConte
       context: MetricsContext
   ): Unit = update(duration.toNanos, TimeUnit.NANOSECONDS)
 }
+object OpenTelemetryTimer {
+
+  private[opentelemetry] val TimerUnit: String = "seconds"
+  val TimerUnitAndSuffix: MetricName = MetricName("duration", TimerUnit)
```
Comment (miklos-da) [inline ledger/metrics/src/main/scala/com/daml/metrics/api/opentelemetry/OpenTelemetryFactory.scala:185]: :+1:

File: ledger/metrics/src/main/scala/com/daml/metrics/api/opentelemetry/OpenTelemetryFactory.scala (hunk 2 above)
Comment (miklos-da) [reply ledger/metrics/src/main/scala/com/daml/metrics/api/opentelemetry/OpenTelemetryFactory.scala:187]: Or
```suggestion
  private val NanosecondsPerSecond = 1_000_000_000
```

**Review by miklos-da — APPROVED** [review]

**Review by lucb-da — APPROVED** [review]
Tested with the json-api metrics, it works fine.
It generates `daml_http_json_api_requests_duration_seconds_duration_seconds_bucket` with double `duration` for the latency, but I am already fixing this in my next PR.

**Pull request merged**